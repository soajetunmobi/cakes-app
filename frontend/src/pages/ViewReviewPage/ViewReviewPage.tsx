import React, { FunctionComponent } from 'react'
import { useHistory } from 'react-router-dom'
import { LoadingIndicator } from '../../components/LoadingIndicator'
import { ErrorPanel } from '../../components/ErrorPanel'
import { routes } from '../../common/const/routes'
import { useFetchCakeReview } from '../../hooks'
import { IHistory } from '../../types/types'

export const ViewReviewPage: FunctionComponent = () => {
  const { review, isLoading, error } = useFetchCakeReview()
  const history = useHistory<IHistory>()

  if (isLoading || !review) {
    return <LoadingIndicator />
  }

  if (error) {
    const errorMessage = <div>{error}</div>
    return <ErrorPanel children={errorMessage} />
  }

  function goBackToCakeList() {
    history.push(routes.cakes)
  }

  return (
    <div className='form-wrapper'>
      <h2 className='page__title' data-testid='page-title'>
        Cake Review
      </h2>
      <div className='review'>
        <div className='row'>
          <span className='cake__image'>
            <img alt={review.name} data-testid='image' src={review.imageUrl} />
          </span>
        </div>
        <div className='row'>
          <span className='cake__name' data-testid='name'>
            {review.name}
          </span>
        </div>
        <div className='row'>
          <span className='cake__comment' data-testid='comment'>
            {review.comment}
          </span>
        </div>

        <div className='row'>
          <label>Yum factor</label>
          <div data-testid='yum-factor-wrapper'>
            <span id='1' className={`circle${review.yumFactor === 1 ? ' active' : ''}`}>
              1
            </span>
            <span id='2' className={`circle${review.yumFactor === 2 ? ' active' : ''}`}>
              2
            </span>
            <span id='3' className={`circle${review.yumFactor === 3 ? ' active' : ''}`}>
              3
            </span>
            <span id='4' className={`circle${review.yumFactor === 4 ? ' active' : ''}`}>
              4
            </span>
            <span id='5' className={`circle${review.yumFactor === 5 ? ' active' : ''}`}>
              5
            </span>
          </div>
        </div>
        <div className='buttons-container buttons-container--flex-end'>
          <button onClick={() => goBackToCakeList()} className='button--primary'>
            Back
          </button>
        </div>
      </div>
    </div>
  )
}
