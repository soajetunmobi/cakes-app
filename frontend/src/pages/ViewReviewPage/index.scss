.row{
  width: 100%;
  position: relative;
  margin-bottom: 37px;
  text-align: center;
}
