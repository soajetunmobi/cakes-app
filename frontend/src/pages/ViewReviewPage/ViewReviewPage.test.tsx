import React from 'react'
import renderer from 'react-test-renderer'
import { ViewReviewPage } from './ViewReviewPage'

describe('ViewReviewPage component', () => {
  it('should render correctly', () => {
    const reviewPage = renderer.create(<ViewReviewPage />).toJSON()
    expect(reviewPage).toMatchSnapshot()
  })
})
