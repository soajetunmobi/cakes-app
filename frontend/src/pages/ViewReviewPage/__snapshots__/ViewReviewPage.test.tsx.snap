// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewReviewPage component should render correctly 1`] = `
<div
  className="loader"
/>
`;
